File: python_src/game/inventory.py
from rich.table import Table

from game.utils.ui import c


class Inventory:
    def __init__(self) -> None:
        self.items = {}

    def add_item(self, item: str, quantity=1) -> None:
        """Add item to inventory."""
        if item in self.items:
            self.items[item] += quantity
        else:
            self.items[item] = quantity
        c.print(f"[bold green]Added {quantity}x {item} to inventory.[/bold green]")

    def remove_item(self, item: str, quantity=1) -> None:
        """Remove item from inventory."""
        if item in self.items:
            if self.items[item] > quantity:
                self.items[item] -= quantity
                c.print(f"[bold yellow]Removed {quantity}x {item} from invrntory.")
            else:
                del self.items[item]
                c.print(f"[bold red]{item} removed from inventory.")
        else:
            c.print(f"[bold red]{item} not found in inventory.")

    def show_inventory(self) -> None:
        """Show inventory in table."""
        if not self.items:
            c.print("[bold red]Your inventory is empty.[/bold red]")
            return

        table = Table(title="[bold cyan]Inventory")
        table.add_column("Item", style="bold yellow")
        table.add_column("Quantity", justify="right", style="bold green")

        for item, quantity in self.items.items():
            table.add_row(item, str(quantity))

        c.print(table)

    def to_dict(self) -> None:
        """Save inventory into dictionary."""
        return self.items

    def from_dict(self, data) -> None:
        """Load inventory from dictionary."""
        self.items = data

File: src/game/inventory.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;
use crate::utils::ui::print_color;

#[derive(Serialize, Deserialize, Clone, Debug)]
pub struct Inventory {
    pub items: HashMap<String, i32>,
}

impl Inventory {
    pub fn new() -> Self {
        Self {
            items: HashMap::new(),
        }
    }

    pub fn add_item(&mut self, item: &str, quantity: i32) {
        let count = self.items.entry(item.to_string()).or_insert(0);
        *count += quantity;
        print_color(&format!("Added {}x {} to inventory.", quantity, item), "green");
    }

    pub fn remove_item(&mut self, item: &str, quantity: i32) {
        if let Some(count) = self.items.get_mut(item) {
            if *count > quantity {
                *count -= quantity;
                print_color(&format!("Removed {}x {} from inventory.", quantity, item), "yellow");
            } else {
                self.items.remove(item);
                print_color(&format!("{} removed from inventory.", item), "red");
            }
        } else {
            print_color(&format!("{} not found in inventory.", item), "red");
        }
    }

    pub fn show_inventory(&self) {
        if self.items.is_empty() {
            print_color("Your inventory is empty.", "red");
            return;
        }

        println!("--- Inventory ---");
        for (item, quantity) in &self.items {
            println!("{}: {}", item, quantity);
        }
        println!("-----------------");
    }
}
